### PR TITLE
[prometheus] Update Helm dependency to use new kube-state-metrics chart versions

### PR DESCRIPTION
**Pull request by chris-vest**
#### What this PR does / why we need it:

1) Updates the version of the `kube-state-metrics` dependency to `"2.9.*"` - this will allow for support of using the latest features of the chart

#### Which issue this PR fixes
None.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by torstenwalter — APPROVED** [review]

**Comment by chris-vest** [comment]
@torstenwalter Looks like their new Helm repo hasn't been set up properly.

**Comment by chris-vest** [comment]
```➜ helm repo add kube-state-metrics https://kubernetes.github.io/kube-state-metrics
Error: looks like "https://kubernetes.github.io/kube-state-metrics" is not a valid chart repository or cannot be reached: failed to fetch https://kubernetes.github.io/kube-state-metrics/index.yaml : 404 Not Found```

**Review by torstenwalter — CHANGES_REQUESTED** [review]
CI reports:

 `Error: could not find : chart kube-state-metrics not found in https://kubernetes.github.io/kube-state-metrics`.

You need to add the repository to [ct.yaml](https://github.com/prometheus-community/helm-charts/blob/main/ct.yaml#L6-L9) to fix it.

```
chart-repos:
  - stable=https://charts.helm.sh/stable
  - prometheus-community=https://prometheus-community.github.io/helm-charts
  - grafana=https://grafana.github.io/helm-charts
  - kube-state-metrics=https://kubernetes.github.io/kube-state-metrics
```

**Comment by chris-vest** [comment]
@torstenwalter Locally it doesn't work either, the `kube-state-metrics` new Chart repo doesn't have the `index.yaml` file. 

I have pivoted to just updating the allowed dependency version to `2.9.*` and not changing the chart location.

**Comment by torstenwalter** [comment]
@chris-vest You are right. I just double checked it. The `gh-pages` branch is still missing in that repository and the first version still needs to be published.

**Comment by chris-vest** [comment]
> @chris-vest You are right. I just double checked it. The `gh-pages` branch is still missing in that repository and the first version still needs to be published.

I'm happy to monitor it and do another PR to update to the new location once they've set that up properly.

**Comment by torstenwalter** [comment]
https://github.com/kubernetes/kube-state-metrics/pull/1326

**Comment by chris-vest** [comment]
> [kubernetes/kube-state-metrics#1326](https://github.com/kubernetes/kube-state-metrics/pull/1326)

I've reverted to use the new repo again.

**Comment by monotek** [comment]
https://kubernetes.github.io/kube-state-metrics/index.yaml brings a 404.
Seems repo needs some more fixes: https://github.com/kubernetes/kube-state-metrics/pull/1329

**Comment by scottrigby** [comment]
Hang tight. We now know this was due to automated settings controlled by prow (for anyone curious, this is it: https://github.com/kubernetes/kube-state-metrics/pull/1334#issuecomment-748113824). This PR should fix that: https://github.com/kubernetes/test-infra/pull/20291

We'll then need to make another PR like https://github.com/kubernetes/kube-state-metrics/pull/1334 again. Unless there are other issues, this should add and index the next release.

**Comment by scottrigby** [comment]
Follow-up PR https://github.com/kubernetes/test-infra/pull/20322 ⏳

**Comment by mrueg** [comment]
@chris-vest @monotek @torstenwalter thanks to @scottrigby the repo has been set up now: https://kubernetes.github.io/kube-state-metrics/ 

@chris-vest can you fix the merge conflicts on your PR?

**Comment by chris-vest** [comment]
@mrueg All done!

**Review by torstenwalter — DISMISSED** [review]

**Comment by torstenwalter** [comment]
Now it just needs an approval from one of the chart maintainers and we are good to go :100:

**Review by naseemkullah — DISMISSED** [review]

**Comment by torstenwalter** [comment]
@naseemkullah do you want to merge it or does it need another review from you opinion?

**Comment by naseemkullah** [comment]
@torstenwalter I know @monotek recently change ksm related stuff so thought he might want to have a look, but then again we have the required 2 approvers, let's merge by EOD.

**Comment by chris-vest** [comment]
@torstenwalter @naseemkullah Sorry guys,  I just pushed up a change to the `values.yaml` in order to provide the right link to the chart - it'll need another review.

**Review by torstenwalter — APPROVED** [review]

File: ct.yaml
Comment (naseemkullah) [inline ct.yaml:11]: Are bitnami charts still required anywhere else in the repo? If not it could be removed from here.

**Review by naseemkullah — APPROVED** [review]

**Pull request merged**
